src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -100,16 +100,20 @@ private static IEnumerable<string> GetDotNetBasePaths(string workingDirectory)
             foreach (string dir in Environment.GetEnvironmentVariable("PATH").Split(Path.PathSeparator))
             {
                 string filePath = Path.Combine(dir, exeName);
-                if (File.Exists(Path.Combine(dir, exeName)))
+                if (File.Exists(filePath))
                 {
-                    dotnetPath = filePath;
-                    break;
+                    filePath = Path.GetDirectoryName(isWindows ? filePath : realpath(filePath) ?? filePath);
```

Comment:
I don't think I understand the `isWindows` check here. You can have dangling symlinks on Windows just as much, right?

Reply:
That is an excellent question. You can in theory, but I don't think I've ever seen someone who has a path to a symlink to dotnet.exe on their path rather than dotnet.exe itself on windows. The real reason I did this for unix and not windows is that that's what the SDK does like here:

https://github.com/dotnet/sdk/blob/733eec58c350f139bbedbdccb38df128ca60b8c3/src/Resolvers/Microsoft.DotNet.NativeWrapper/EnvironmentProvider.cs#L80-L85

Since I used the SDK as a reference when initially trying to understand how to call hostfxr, I just went with it.